crates/holochain/src/conductor/conductor/tests.rs

```diff
@@ -275,3 +279,231 @@ async fn proxy_tls_inner(
+    let bad_zome =
+        InlineZome::new_unique(Vec::new()).callback("validate", |_api, _data: ValidateData| {
+            panic!("intentional panic during validation");
+            #[allow(unreachable_code)]
+            Ok(ValidateResult::Valid)
+        });
+
+    let (conductor, _) = common_genesis_test_app(bad_zome).await;
+
+    // - Ensure that the app was removed and both Cells were uninstalled
+    //   because one Cell panicked during Genesis
+    assert_eq_retry_10s!(
+        {
+            let state = conductor.get_state_from_handle().await.unwrap();
+            (state.active_apps.len(), state.inactive_apps.len())
+        },
+        (0, 0)
+    );
+}
+
+#[tokio::test(flavor = "multi_thread")]
+async fn test_cells_self_destruct_on_bad_validation_during_genesis() {
+    observability::test_run().ok();
+    let bad_zome =
+        InlineZome::new_unique(Vec::new()).callback("validate", |_api, _data: ValidateData| {
+            Ok(ValidateResult::Invalid(
+                "intentional invalid result for testing".into(),
+            ))
+        });
+
+    let (conductor, _) = common_genesis_test_app(bad_zome).await;
+
+    assert_eq_retry_10s!(
+        {
+            let state = conductor.get_state_from_handle().await.unwrap();
+            (state.active_apps.len(), state.inactive_apps.len())
+        },
+        (0, 0)
+    );
+}
+
+#[tokio::test(flavor = "multi_thread")]
+async fn test_bad_entry_validation_after_genesis_returns_zome_call_error() {
+    observability::test_run().ok();
+    let unit_entry_def = EntryDef::default_with_id("unit");
+    let bad_zome = InlineZome::new_unique(vec![unit_entry_def.clone()])
+        .callback("validate_create_entry", |_api, _data: ValidateData| {
+            Ok(ValidateResult::Invalid(
+                "intentional invalid result for testing".into(),
+            ))
+        })
+        .callback("create", move |api, ()| {
+            let entry_def_id: EntryDefId = unit_entry_def.id.clone();
+            let entry = Entry::app(().try_into().unwrap()).unwrap();
+            let hash = api.create(EntryWithDefId::new(entry_def_id, entry))?;
+            Ok(hash)
+        });
+
+    let (conductor, app) = common_genesis_test_app(bad_zome).await;
+
+    let (_, cell_bad) = app.into_tuple();
+
+    let result: ConductorApiResult<HeaderHash> = conductor
+        .call_fallible(&cell_bad.zome("bad"), "create", ())
+        .await;
+
+    // - The failed validation simply causes the zome call to return an error
+    assert_matches!(result, Err(_));
+
+    // - The app is not deactivated
+    assert_eq_retry_10s!(
+        {
+            let state = conductor.get_state_from_handle().await.unwrap();
+            (state.active_apps.len(), state.inactive_apps.len())
+        },
+        (1, 0)
+    );
+}
+
+#[tokio::test(flavor = "multi_thread")]
+#[ignore = "Seems we want to test this, but actually it seems that panics in entry creation validation are unrecoverable"]
```

Comment:
Hmmm will this cause a problem for hosted happs when lair gets disconnected part way through entry creation, or is this only inside the validation routine?